pkg/component/runtime/manager_test.go

```diff
@@ -482,6 +482,167 @@ func TestManager_FakeInput_Features(t *testing.T) {
 	}
 }
 
+func TestManager_FakeInput_Limits(t *testing.T) {
```

Comment:
This is a great test, can you add another one but with a shipper enabled. It would be great to verify that the shipper and the components get the limits.

Reply:
Addressed in https://github.com/elastic/elastic-agent/pull/3179/commits/9b007bbcf0707e4935f62e82b0178160304cbfe8